include/boost/multiprecision/detail/functions/pow.hpp

```diff
@@ -527,12 +523,14 @@ typename std::enable_if<should_use_log_agm<T>::value>::type eval_log(T& result_x
   // lexical_cast would do the trick.
```

Comment:
I used a slightly different approach based on the exponent of the epsilon threshold. Does this look reasonable to you?